src/playlist-loader.js

```diff
@@ -421,7 +421,7 @@ export default class PlaylistLoader extends EventTarget {
     this.dateRangesStorage_.setPendingDateRanges(mediaPlaylist.dateRanges);
     const availableDateRanges = this.dateRangesStorage_.getDateRangesToProcess();
 
-    if (!availableDateRanges.length) {
+    if (!availableDateRanges.length || !this.addDateRangesToTextTrack_) {
```

Comment:
Noticed that there were several `!this.addDateRangesToTextTrack_ is not a function` errors when I tested dateRanges with a specific stream. And noticed that the function altered between being undefined and being available on each `loadedplaylist` event (line 410). Not sure why this is happening, so added a quick fix before debugging in detail.

Reply:
This seems to be because the test stream has subtitles, which in this case we create a `PlaylistLoader` for, this does not have the `addDateRangesToTextTrack_` function defined, as it is only needed for the `mainPlaylistLoader` for HLS.